tests/feature-store/test_feature_store.py

```diff
@@ -155,36 +157,102 @@ def test_feature_set_db():
     assert len(sets) == 1, "bad number of results"
 
     feature_set = db.get_feature_set(name)
-    assert feature_set.metadata.name == name, "bas feature set response"
+    assert feature_set.metadata.name == name, "bad feature set response"
 
 
 @pytest.mark.skipif(not has_db(), reason="no db access")
 def test_serverless_ingest():
     init_store()
+    key = "patient_id"
 
-    measurements = fs.FeatureSet("measurements", entities=[Entity("patient_id")])
-    src_df = pd.read_csv(local_dir + "testdata.csv")
-    df = fs.infer_metadata(
-        measurements,
-        src_df,
-        timestamp_key="timestamp",
-        options=fs.InferOptions.default(),
+    measurements = fs.FeatureSet(
+        "measurements", entities=[Entity(key)], timestamp_key="timestamp"
     )
-    print(df.head(5))
     target_path = os.path.relpath(results_dir + "mycsv.csv")
     source = CSVSource("mycsv", path=os.path.relpath(local_dir + "testdata.csv"))
     targets = [CSVTarget("mycsv", path=target_path)]
     if os.path.exists(target_path):
         os.remove(target_path)
 
-    run_ingestion_task(
+    run_ingestion_job(
         measurements,
         source,
         targets,
         name="test_ingest",
-        infer_options=fs.InferOptions.Null,
+        infer_options=fs.InferOptions.schema() + fs.InferOptions.Stats,
         parameters={},
         function=None,
         local=True,
     )
     assert os.path.exists(target_path), "result file was not generated"
+    features = sorted(measurements.spec.features.keys())
+    stats = sorted(measurements.status.stats.keys())
+    print(features)
+    print(stats)
+    stats.remove("timestamp")
+    assert features == stats, "didnt infer stats for all features"
+
+    print(measurements.to_yaml())
+
+
+def prepare_feature_set(name: str, entity: str, data: pd.DataFrame, timestamp_key=None):
+    df_source = fs.sources.DataFrameSource(data, entity, timestamp_key)
+
+    feature_set = fs.FeatureSet(
+        name, entities=[fs.Entity(entity)], timestamp_key=timestamp_key
+    )
+    feature_set.set_targets()
+    df = fs.ingest(feature_set, df_source, infer_options=fs.InferOptions.default())
+    return feature_set, df
+
+
+@pytest.mark.skipif(not has_db(), reason="no db access")
```

Comment:
Doesn't have to be part of this PR, but I've added an integration tests (SDK + API) infrastructure, currently all of these tests doesn't run automatically, you should move all of them there, it's under `tests/integration/sdk_api`

Reply:
will do in the next PR